internal/benchrunner/runners/stream/runner.go

```diff
@@ -120,15 +126,55 @@ func (r *runner) setUp() error {
 	}
 	r.scenarios = scenarios
 
-	if err = r.installPackage(); err != nil {
-		return fmt.Errorf("error installing package: %w", err)
-	}
-
 	err = r.collectGenerators()
 	if err != nil {
 		return fmt.Errorf("can't initialize generator: %w", err)
 	}
 
+	return nil
+}
+
+func (r *runner) validateGenerators() error {
+	for scenarioName, generator := range r.generators {
+		for i := 0; i < numberOfEvents; i++ {
+			buf := bytes.NewBufferString("")
+			err := generator.Emit(buf)
+			if err != nil {
+				return fmt.Errorf("[%s] error while generating event: %w", scenarioName, err)
+			}
+			// check whether the generated event is valid json
+			var event map[string]any
+			err = json.Unmarshal(buf.Bytes(), &event)
+			if err != nil {
+				return fmt.Errorf("[%s] failed to unmarshal json event: %w, generated output: %s", scenarioName, err, buf.String())
+			}
```

Comment:
It would be also nice to validate the fields, as we do in tests and when validating the sample documents, for example here: https://github.com/elastic/elastic-package/blob/a3e3a802b2d2c3f933c4e151f32cf12d5945005f/internal/testrunner/runners/static/runner.go#L117

Reply:
I looked into this, but the fields validator partially depends on test configurations like `numeric_keyword_fields` which might not be available / can't be matched properly to the benchmark config.

I would like to split that part out of this PR if you don't mind.